### ISSUE TITLE
-Wsection in net/rds/ib_stats.c

### ISSUE THREAD

**Issue by tpimh** [ClangBuiltLinux/linux#114]
```
net/rds/ib_stats.c:40:1: warning: section does not match previous declaration [-Wsection]
DEFINE_PER_CPU_SHARED_ALIGNED(struct rds_ib_statistics, rds_ib_stats);
^
./include/linux/percpu-defs.h:144:2: note: expanded from macro 'DEFINE_PER_CPU_SHARED_ALIGNED'
        DEFINE_PER_CPU_SECTION(type, name, PER_CPU_SHARED_ALIGNED_SECTION) \
        ^
./include/linux/percpu-defs.h:93:9: note: expanded from macro 'DEFINE_PER_CPU_SECTION'
        extern __PCPU_ATTRS(sec) __typeof__(type) name;                 \
               ^
./include/linux/percpu-defs.h:49:26: note: expanded from macro '__PCPU_ATTRS'
        __percpu __attribute__((section(PER_CPU_BASE_SECTION sec)))     \
                                ^
net/rds/ib.h:446:1: note: previous attribute is here
DECLARE_PER_CPU(struct rds_ib_statistics, rds_ib_stats);
^
./include/linux/percpu-defs.h:113:2: note: expanded from macro 'DECLARE_PER_CPU'
        DECLARE_PER_CPU_SECTION(type, name, "")
        ^
./include/linux/percpu-defs.h:87:9: note: expanded from macro 'DECLARE_PER_CPU_SECTION'
        extern __PCPU_ATTRS(sec) __typeof__(type) name
               ^
./include/linux/percpu-defs.h:49:26: note: expanded from macro '__PCPU_ATTRS'
        __percpu __attribute__((section(PER_CPU_BASE_SECTION sec)))     \
                                ^
1 warning generated.
```

**Comment by nathanchance** [comment]
Patch sent: https://lore.kernel.org/lkml/20180921180451.18711-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Patch v2: https://lore.kernel.org/lkml/20180923032938.27093-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Unfortunately v2 didn't make it to David Miller in time so it's now in two patches

https://git.kernel.org/pub/scm/linux/kernel/git/davem/net.git/commit/?id=8360ed6745df6de2d8ddff8e2116789258fe5890

https://git.kernel.org/pub/scm/linux/kernel/git/davem/net.git/commit/?id=16fdf8ba98391650ce4bc4f3f71629d8a413bc21

**Comment by nathanchance** [comment]
Merged into mainline, will be in 4.19-rc6:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8360ed6745df6de2d8ddff8e2116789258fe5890

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=16fdf8ba98391650ce4bc4f3f71629d8a413bc21
